File: services-java/va-service/test_transcribe.py
"""
Simple Python test for Transcribe RPC
Run: python test_transcribe.py
"""
import grpc
import sys
from google.protobuf import timestamp_pb2

# Import generated proto files
sys.path.insert(0, 'target/generated-sources/protobuf/python')
try:
    import voice_pb2
    import voice_pb2_grpc
except ImportError:
    print("❌ Proto files not found. Need to generate them first.")
    print("   Run: pip install grpcio grpcio-tools protobuf")
    print("   Then: python -m grpc_tools.protoc -I=src/main/proto --python_out=target/generated-sources/protobuf/python --grpc_python_out=target/generated-sources/protobuf/python src/main/proto/voice.proto")
    sys.exit(1)

def test_transcribe():
    print("═" * 60)
    print("  Testing va-service Transcribe RPC")
    print("═" * 60)
    print()
    
    # Create channel
    channel = grpc.insecure_channel('localhost:50051')
    stub = voice_pb2_grpc.VoiceServiceStub(channel)
    
    # Create dummy audio data
    dummy_audio = b'test audio data - placeholder'
    
    # Create request
    request = voice_pb2.AudioChunk(
        session_id=f'test-session-python',
        audio_data=dummy_audio,
        format='webm',
        timestamp=1234567890,
        sequence_number=1,
        customer_id='test-customer',
        is_final_chunk=True
    )
    
    print("Sending Transcribe request:")
    print(f"  Session ID: {request.session_id}")
    print(f"  Format: {request.format}")
    print(f"  Audio size: {len(dummy_audio)} bytes")
    print()
    
    try:
        # Make RPC call
        response = stub.Transcribe(request, timeout=30)
        
        print("✅ Transcribe Response:")
        print(f"  Session ID: {response.session_id}")
        print(f"  Text: {response.text}")
        print(f"  Confidence: {response.confidence}")
        print(f"  Language: {response.language}")
        print(f"  Is Final: {response.is_final}")
        if response.metadata:
            print(f"  Provider: {response.metadata.provider}")
            print(f"  Model: {response.metadata.model}")
        print()
        print("✅ Test PASSED!")
        return True
        
    except grpc.RpcError as e:
        print(f"❌ Error: {e.code()}")
        print(f"   Details: {e.details()}")
        return False
    finally:
        channel.close()

if __name__ == '__main__':
    success = test_transcribe()
    sys.exit(0 if success else 1)

File: services-java/va-service/test-transcribe.js
// Quick test for Transcribe RPC using grpc-js
// Run: node test-transcribe.js

const grpc = require('@grpc/grpc-js');
const protoLoader = require('@grpc/proto-loader');
const path = require('path');
const fs = require('fs');

// Load proto
const PROTO_PATH = path.join(__dirname, 'src', 'main', 'proto', 'voice.proto');
const packageDefinition = protoLoader.loadSync(PROTO_PATH, {
    keepCase: true,
    longs: String,
    enums: String,
    defaults: true,
    oneofs: true
});

const voiceProto = grpc.loadPackageDefinition(packageDefinition).com.ai.va.grpc;

// Create client
const client = new voiceProto.VoiceService(
    'localhost:50051',
    grpc.credentials.createInsecure()
);

// Test Transcribe RPC
function testTranscribe() {
    console.log('Testing Transcribe RPC...\n');
    
    // Create dummy audio data (in real scenario, this would be actual audio bytes)
    const dummyAudio = Buffer.from('test audio data - this is just a placeholder');
    
    const request = {
        session_id: 'test-session-' + Date.now(),
        audio_data: dummyAudio,
        format: 'webm',
        timestamp: Date.now(),
        sequence_number: 1,
        customer_id: 'test-customer',
        is_final_chunk: true
    };
    
    console.log('Sending Transcribe request:');
    console.log('  Session ID:', request.session_id);
    console.log('  Format:', request.format);
    console.log('  Audio size:', dummyAudio.length, 'bytes');
    console.log('');
    
    client.transcribe(request, (error, response) => {
        if (error) {
            console.error('❌ Error:', error.message);
            console.error('   Code:', error.code);
            console.error('   Details:', error.details);
            process.exit(1);
        } else {
            console.log('✅ Transcribe Response:');
            console.log('  Session ID:', response.session_id);
            console.log('  Text:', response.text);
            console.log('  Confidence:', response.confidence);
            console.log('  Language:', response.language);
            console.log('  Is Final:', response.is_final);
            if (response.metadata) {
                console.log('  Provider:', response.metadata.provider);
                console.log('  Model:', response.metadata.model);
            }
            console.log('\n✅ Test PASSED!');
            process.exit(0);
        }
    });
}

// Run test
console.log('═══════════════════════════════════════');
console.log('  Testing va-service Transcribe RPC    ');
console.log('═══════════════════════════════════════\n');

testTranscribe();

// Timeout after 30 seconds
setTimeout(() => {
    console.error('❌ Test timed out after 30 seconds');
    process.exit(1);
}, 30000);
